### DISCUSSION TITLE
Delay for updated entities

### DISCUSSION THREAD

**Discussion by xnisayx** [humbertogontijo/homeassistant-roborock#483] [Q&A]
How quickly do the settings change when you change the suction power, for example?

When I change the suction power, the change is made immediately on the robot.
 However, the current settings are only displayed correctly after at least 1 - 2 minutes.

Is this normal?

![Screenshot_20230920-142138.png](https://github.com/humbertogontijo/homeassistant-roborock/assets/50765521/2358b092-e9d9-49d5-9e05-9ad74eb1e8ec)

**Comment by manunited10** [comment]
I also have an issue with delay of updating `water_box_attached` in Roborock S7. It takes 1-2 minutes to show that I removed it.
Would be amazing if it could be looked at.

TLDR
Actually what I liked to have was `dustbin` removed/attached sensor to figure out if it has been emptied or not. So I used `water_box_attached` sensor (and told the family to each time emptying the dustbin also refill the waterbox) which is not ideal, but worked for me. Since I migrated to this repo (which works really great on so many sensors) the automation that looks at the `waterbox` is not working properly.

**Comment by modo1302** [comment]
Same here. Sometimes the status of the vacuum actually takes so long to update, that my automation misses the 'returnig to base' status. The vacuum already manages to return to the docking station, and when the status updates - it's already charging.
